api/lib/controllers/meeting.js

```diff
@@ -199,6 +200,80 @@ module.exports = {
     const participants = await Participant.find({ meeting_id: _id });
 
     return res.status( 200 ).send( participants );
+  },
+
+  async getAllMeetings( req, res ) {
+    const subject_email = req.credentials.user.email;
+    const subject_id = req.credentials.sub;
+    const { limit = 0, skip = 0 } = req.query; // Thanks tom
+
+    try {
+      const pipeline = [
+        {
+          $match: { _id: ObjectID( subject_id ) }
+        },
+        {
+          $lookup: {
+            from: 'meetings',
+            localField: '_id',
+            foreignField: 'owner_id',
+            as: 'owned_meetings'
+          }
+        },
+        {
+          $lookup: {
+            from: 'participants',
+            pipeline: [
+              { $match: { email: subject_email } },
+              {
+                $lookup: {
+                  from: 'meetings',
+                  localField: 'meeting_id',
+                  foreignField: '_id',
+                  as: 'meetings'
+                }
+              },
+              {
+                $project: {
+                  meeting: { $arrayElemAt: [ '$meetings', 0 ] }
+                }
+              },
+              {
+                $replaceRoot: {
+                  newRoot: '$meeting'
+                }
+              }
+            ],
+            as: 'participating_meetings'
+          }
+        },
+        {
+          $project: {
+            meetings: {
+              $concatArrays: [ '$participating_meetings', '$owned_meetings' ]
+            }
+          }
+        },
+        { $unwind: '$meetings' },
+        { $sort: { 'meetings.createdAt': -1 } },
+        { $skip: parseInt( skip ) },
+        { $limit: parseInt( limit ) },
+        {
+          $group: {
+            _id: '$_id',
+            meetings: { $push: '$meetings' }
+          }
+        }
```

Comment:
If you swap this part out with this:
```js
        { $unwind: '$meetings' },
        { $replaceRoot: { newRoot: '$meetings' } },
        { $sort: { 'date': -1 } },
        { $skip: parseInt( skip ) },
        { $limit: parseInt( limit ) }
```
The `replateRoot` operation will make the `meetings` doc for each of the duplicated user docs from the `unwind` the root doc. Then we just get out an array of the meetings.